rest_framework/fields.py

```diff
@@ -1106,7 +1106,7 @@ def to_internal_value(self, data):
         return data
 
     def to_representation(self, value):
-        if self.use_url:
+        if self.use_url and hasattr(value, 'url'):
             if not value:
```

Comment:
Looks like this `if not value` check should be moved to the top of the method instead of being hidden away here.